app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_list/AgendaViewModel.kt

```diff
@@ -148,4 +159,14 @@ class AgendaViewModel @Inject constructor(
             }
         }
     }
+
+    private fun observeConnectivity() {
+        viewModelScope.launch {
+            isConnected.collect { connected ->
+                if (connected) {
+                    agendaRepository.syncDeletedAgendaItems()
+                }
+            }
+        }
+    }
```

Comment:
Syncing each time the user regains connection may either be too frequent or too infrequent 😅 
How about performing this sync when the user lands on the `AgendaListScreen`?